File: leetcode/arrays/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int ans[] = new int[2];
        HashSet<Integer> map = new HashSet<>();
        for(int i=0; i<nums.length; i++){
            if(map.contains(target-nums[i])){
                ans[0]=i;
                break;
            }
            map.add(nums[i]);
        }
        for(int i=0;i<nums.length;i++){
            if(nums[i]==target-nums[ans[0]]){
                ans[1]=i;
                break;
            }
        }
        return ans;
    }
}

File: interviewBit/level 5/twoSum.cpp
vector<int> Solution::twoSum(const vector<int> &A, int B)
{
    unordered_map<int, int> mp;

    for (int i = 0; i < A.size(); i++)
    {

        if (mp.find(B - A[i]) != mp.end())
        {

            return {mp[B - A[i]] + 1, i + 1};
        }

        if (mp.find(A[i]) == mp.end())
            mp[A[i]] = i;
    }

    return {};
}
